src/pj_fwd.c

```diff
@@ -81,9 +109,42 @@ PJ_COORD pj_fwd_finalize (PJ *P, PJ_COORD coo) {
     }
 
     /* Handle false eastings/northings and non-metric linear units */
-    coo.xyz.x = P->fr_meter  * (coo.xyz.x + P->x0);
-    coo.xyz.y = P->fr_meter  * (coo.xyz.y + P->y0);
-    coo.xyz.z = P->vfr_meter * (coo.xyz.z + P->z0);
+    if (P->right==PJ_IO_UNITS_CARTESIAN) {
+        coo.xyz.x = P->fr_meter * (coo.xyz.x + P->x0);
+        coo.xyz.y = P->fr_meter * (coo.xyz.y + P->y0);
+        coo.xyz.z = P->fr_meter * (coo.xyz.z + P->z0);
+    }
+    if (P->right==PJ_IO_UNITS_PROJECTED || P->right==PJ_IO_UNITS_CLASSIC) {
+        coo.xyz.x = P->fr_meter  * (coo.xyz.x + P->x0);
+        coo.xyz.y = P->fr_meter  * (coo.xyz.y + P->y0);
+        coo.xyz.z = P->vfr_meter * (coo.xyz.z + P->z0);
+    }
+
+    if (P->right==PJ_IO_UNITS_RADIANS && P->left!=PJ_IO_UNITS_RADIANS) {
+        /* Distance from central meridian, taking system zero meridian into account */
+        coo.lp.lam = coo.lp.lam + P->from_greenwich + P->lam0;
+
+         /* adjust longitude to central meridian */
+        if (0==P->over)
+            coo.lpz.lam = adjlon(coo.lpz.lam);
+
+        if (P->vgridshift)
+            coo = proj_trans (P->vgridshift, -1, coo);
```

Comment:
Use `PJ_INV` here. And many other places in the code below.

Reply:
I don't agree. `PJ_INV/PJ_FWD` were introduced for your convenience. I actually find it quite a lot more readable to have negative mean backward, and positive mean forward.

And since there is more than enough of layered negations in this code, I really prefer the concise to the verbose here.

Reply:
I guess we will just have to agree to disagree here :-)

Reply:
Sounds very agreeable :-)

Reply:
That being said, if you find it unreadable, I would not mind switching to `PJ_FWD/PJ_INV` once we have established that the material works as intended (last step before merging).

It's just that I find the `PJ_FWD/PJ_INV` convention less readable than the mathematical convention of +1/-1, and this stuff is pretty obnoxious already.